firefox-ios/Client/TabManagement/TabManagerImplementation.swift

```diff
@@ -414,7 +414,10 @@ class TabManagerImplementation: LegacyTabManager, Notifiable, WindowSimpleTabsPr
         if selectedTab?.isPrivate ?? false {
             _selectedIndex = -1
         }
-        privateTabs.forEach { $0.close() }
+        privateTabs.forEach { tab in
+            tab.close()
+            delegates.forEach { $0.get()?.tabManager(self, didRemoveTab: tab, isRestoring: false) }
```

Comment:
Don't know if this is actually the ideal fix, still need to see when/how these delegate methods are actually meant to be called. This does appear to fix the crash for me however (which I can repro 100% so far). The underlying problem AFAICS is that we don't seem to be updating the data store during tab removal when clearing all private tabs.

Would be nice to get 👀 from someone who is better versed with our tab code however. cc @OrlaM @nbhasin2

Reply:
I double checked and yes you are right we were not calling the didRemove anywhere

Reply:
Good catch!